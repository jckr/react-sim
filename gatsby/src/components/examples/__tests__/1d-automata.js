import React from 'react';
import renderer from 'react-test-renderer';
import Automata, { initData, updateData, params } from '../automata';
import Random from '../../helpers/random';

describe('1d automata', () => {
  let data;
  it('init data function works', () => {
    const rng = new Random();
    data = initData(params, rng.get);
    expect(data).toMatchSnapshot();
  });
  it('update function', () => {
    const rule = 110;
    const cases = [
      { input: [1, 1, 1], output: 0 },
      { input: [1, 1, 0], output: 1 },
      { input: [1, 0, 1], output: 1 },
      { input: [1, 0, 0], output: 0 },
      { input: [0, 1, 1], output: 1 },
      { input: [0, 1, 0], output: 1 },
      { input: [0, 0, 1], output: 1 },
      { input: [0, 0, 0], output: 0 },
    ];
    cases.forEach(c => {
      const updatedData = updateData({ data: c.input, params: { rule } });
      if (c.output) {
        expect(updatedData[1]).toBeTruthy();
      } else {
        expect(updatedData[1]).toBeFalsy();
      }
    });
  });
  it('model renders properly', () => {
    const tree = renderer
      .create(<Automata data={data} params={params} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
