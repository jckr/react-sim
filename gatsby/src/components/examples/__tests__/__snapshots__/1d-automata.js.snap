// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1d automata init data function works 1`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;

exports[`1d automata model renders properly 1`] = `
<div
  className="css-caaeif"
>
  <div
    className="css-l3rx45"
  >
    <div
      className="css-4cffwv"
    >
      <div
        className="css-l3rx45"
        data={
          Array [
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ]
        }
        results={Array []}
      >
        <div
          style={
            Object {
              "height": 288,
              "overflow": "hidden",
              "position": "relative",
              "width": 600,
            }
          }
        >
          <div
            className="css-1peke46"
            direction="column"
          />
        </div>
        <div
          className="css-uofa4f"
        >
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-1io72r5"
              />
              <div
                className="css-1io72r5"
              />
              <div
                className="css-1io72r5"
              />
            </div>
            <div
              className="css-1io72r5"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-1io72r5"
              />
              <div
                className="css-1io72r5"
              />
              <div
                className="css-307zd0"
              />
            </div>
            <div
              className="css-307zd0"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-1io72r5"
              />
              <div
                className="css-307zd0"
              />
              <div
                className="css-1io72r5"
              />
            </div>
            <div
              className="css-307zd0"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-1io72r5"
              />
              <div
                className="css-307zd0"
              />
              <div
                className="css-307zd0"
              />
            </div>
            <div
              className="css-307zd0"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-307zd0"
              />
              <div
                className="css-1io72r5"
              />
              <div
                className="css-1io72r5"
              />
            </div>
            <div
              className="css-1io72r5"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-307zd0"
              />
              <div
                className="css-1io72r5"
              />
              <div
                className="css-307zd0"
              />
            </div>
            <div
              className="css-307zd0"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-307zd0"
              />
              <div
                className="css-307zd0"
              />
              <div
                className="css-1io72r5"
              />
            </div>
            <div
              className="css-307zd0"
            />
          </div>
          <div
            className="css-wm33az"
            onClick={[Function]}
          >
            <div
              className="css-1kw0tju"
            >
              <div
                className="css-307zd0"
              />
              <div
                className="css-307zd0"
              />
              <div
                className="css-307zd0"
              />
            </div>
            <div
              className="css-1io72r5"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-wl408b"
      >
        <div
          className="css-1g78tb9"
        >
          <div
            className="css-l3rx45"
          >
            <label
              className="css-rqmth"
              htmlFor="Rule"
            >
              Rule: 
              110
            </label>
            <div
              className="css-190meoe"
            >
              <div
                className="css-46hvup"
                onClick={[Function]}
              >
                0
              </div>
              <input
                className="css-u6ybtt"
                id="Rule"
                max={255}
                min={0}
                name="Rule"
                onChange={[Function]}
                step={1}
                type="range"
                value={110}
              />
              <div
                className="css-xevc"
                onClick={[Function]}
              >
                255
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-wl408b"
      >
        <div
          className="css-1g78tb9"
        >
          <div
            className="css-e101rj"
          >
            <button
              className="css-1o31xag"
              onClick={[Function]}
            >
              <img
                alt="play_arrow"
                src="https://icon.now.sh/play_arrow/fff"
              />
            </button>
            <button
              className="css-1o31xag"
              onClick={[Function]}
            >
              <img
                alt="reset"
                src="https://icon.now.sh/refresh/fff"
                style={
                  Object {
                    "transform": "scaleX(-1)",
                  }
                }
              />
            </button>
            <button
              className="css-1o31xag"
              onClick={[Function]}
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.72 7.423C10.72 8.523 11.62 9.423 12.72 9.423C13.82 9.423 14.72 8.523 14.72 7.423C14.72 6.323 13.82 5.423 12.72 5.423C11.62 5.423 10.72 6.323 10.72 7.423Z"
                  fill="#fff"
                />
                <path
                  d="M10 7.5L6 3V6H2V9H6V12L10 7.5Z"
                  fill="#fff"
                />
              </svg>
            </button>
            <div
              className="css-l3rx45"
            >
              <label
                className="css-rqmth"
                htmlFor="Time"
              >
                Time: 
                0
              </label>
              <div
                className="css-190meoe"
              >
                <div
                  className="css-46hvup"
                  onClick={[Function]}
                >
                  0
                </div>
                <input
                  className="css-u6ybtt"
                  id="Time"
                  max={100}
                  min={0}
                  name="Time"
                  onChange={[Function]}
                  step={1}
                  type="range"
                  value={0}
                />
                <div
                  className="css-xevc"
                  onClick={[Function]}
                >
                  100
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
